metals/src/main/scala/scala/meta/internal/metals/MetalsLanguageServer.scala

```diff
@@ -1763,13 +1775,74 @@ class MetalsLanguageServer(
                 .applyEdit(new ApplyWorkspaceEditParams(workspaceEdit))
                 .asScala
             } yield ()
-          }
+          })
 
           futureOpt.getOrElse {
             languageClient.showMessage(Messages.InsertInferredTypeFailed)
             Future.unit
           }.withObjectValue
         }
+      case ServerCommands.ExtractMemberDefinition() =>
+        def applyEdits(
+            uri: String,
+            dataOps: Option[CodeActionCommandResult]
+        ): Future[Unit] = {
+          (for {
+            data <- dataOps
+            (edits, newFileUri, range) <- {
+              data match {
+                case ExtractMemberDefinitionResult(
+                      edits,
+                      newFileUri,
+                      range
+                    ) =>
+                  Some((edits, newFileUri, range))
+                case _ => None
+              }
+            }
+            if edits.nonEmpty
+            workspaceEdit = new l.WorkspaceEdit(
+              Map(uri -> edits.asJava).asJava
+            )
+            applyWorkSpaceEditParams = new ApplyWorkspaceEditParams(
+              workspaceEdit
+            )
+          } yield languageClient
+            .applyEdit(applyWorkSpaceEditParams)
+            .asScala
```

Comment:
It could, but the next map uses `newFileUri` and `range` from `ExtractMemberDefinitionResult`, and instead of returning an Option with those data, having to then move again form Option and Futures, I believe it can be better to just map into the new future that returns unit, but any suggestion how to avoid having to pass around `newFileUri` and `range` is welcome